### PR TITLE
Specializer updates

### PR DESCRIPTION
**Pull request by henderiw**
Updates in the specialised
- dealing with ipv4 and ipv6
- fixes bugs

### CODE REVIEW

File: krm-functions/dnn-fn/fn/function.go [hunk 1]
```diff
@@ -190,8 +190,10 @@ func (f *dnnFn) updateDnnResource(dnnObj_ *fn.KubeObject, owned fn.KubeObjects)
 
 func getAnnotations(annotations map[string]string) map[string]string {
 	a := map[string]string{}
-	for k, v := range annotations {
-		a[k] = v
+	if len(annotations) > 0 {
```
Comment (johnbelamaric) [inline krm-functions/dnn-fn/fn/function.go:193]: this shouldn't be needed, range should handle nil and zero length just fine. Why was this necessary?

File: krm-functions/dnn-fn/fn/function.go [hunk 2]
```diff
@@ -190,8 +190,10 @@ func (f *dnnFn) updateDnnResource(dnnObj_ *fn.KubeObject, owned fn.KubeObjects)
...
+		for k, v := range annotations {
+			a[k] = v
+		}
 	}
 	if owner, ok := annotations[condkptsdk.SpecializerPurpose]; ok {
```
Comment (johnbelamaric) [inline krm-functions/dnn-fn/fn/function.go:198]: This may blow up if annotations is nil, though (not 100% on that)

File: krm-functions/dnn-fn/fn/function.go [hunk 3]
```diff
@@ -190,8 +190,10 @@ func (f *dnnFn) updateDnnResource(dnnObj_ *fn.KubeObject, owned fn.KubeObjects)
...
 		a[condkptsdk.SpecializerPurpose] = owner
```
Comment (johnbelamaric) [inline krm-functions/dnn-fn/fn/function.go:199]: didn't we just set this in the loop above?

File: krm-functions/interface-fn/fn/function.go [hunk 4]
```diff
@@ -307,8 +307,10 @@ func (f *itfceFn) getAnnotationsWithvlanClaimName(itfce *nephioreqv1alpha1.Inter
 
 func getAnnotations(annotations map[string]string) map[string]string {
 	a := map[string]string{}
-	for k, v := range annotations {
-		a[k] = v
+	if len(annotations) > 0 {
```
Comment (johnbelamaric) [inline krm-functions/interface-fn/fn/function.go:310]: same point...this seems weird

File: krm-functions/dnn-fn/fn/function.go (hunk 1 above)
Comment (henderiw) [reply krm-functions/dnn-fn/fn/function.go:193]: I did this change and nothing worked any longer so was a bit paranoid and added this, but it did not change anything. So will revert

Comment (johnbelamaric) [reply krm-functions/dnn-fn/fn/function.go:193]: been there, done that :)

**Review by johnbelamaric — APPROVED** [review]
There's a lot here...I gave it a once over but I am relying a lot on the tests.

/approve
/lgtm
